core/CMakeLists.txt

```diff
@@ -256,6 +279,63 @@ set_property(
   APPEND PROPERTY COMPILE_FLAGS -O2
 )
 
+install(DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/../shared/include/userver"
+  DESTINATION "include/"
+  FILES_MATCHING
+  PATTERN "*.hpp"
+  PATTERN "*.h"
+)
+
+install(FILES "${CMAKE_CURRENT_SOURCE_DIR}/src/utils/signal_catcher.hpp"
```

Comment:
copy-pasta: we shouldn't need to edit CMake scripts when adding new source files

Reply:
This part install private file that we need to include after installation (from src/ folder, userver_add_library install only include folder)

Reply:
Stuff in `src/` and `impl` is private API. We provide no support for code that uses such workarounds to access it anyway. If you still need it, move it to public API (`include`). Please expand on why you need it, we'll figure out the best way to expose the necessary tools.